src/Network/GRPC/MQTT/RemoteClient/Session.hs

```diff
@@ -351,5 +353,10 @@ data SessionTopic = SessionTopic
 -- @since 0.1.0.0
 fromRqtTopic :: Topic -> Topic -> Maybe SessionTopic
 fromRqtTopic base topic = do
-  ["grpc", "request", sid, svc, rpc] <- stripPrefix (Topic.split base) (Topic.split topic)
+  ["grpc", "request", sid, escapedSvc, rpc] <- stripPrefix (Topic.split base) (Topic.split topic)
+
+  let unescapeDots :: Text -> Text
+      unescapeDots = Text.map (\c -> if c == '-' then '.' else c)
```

Comment:
This is totally fine and I prefer `if` conditionals to what I'm about to suggest but I did a little code golfing in my head and thought this alternative is also kind of nice:

```haskell
unescapeDots = Text.map \case
  '-' -> '.'
  c -> c
```

Reply:
Yeah, that looks better.